chore(KFLUXVNGD-210): Add source dir to tkn bundle

avi-biton opened the pull request:
- Update recipe.yml to update workspace
- Add `SOURCE_CODE_DIR` to point to the source folder (defaults to `source`)
- Update `tkn-bundle-oci-ta` task

arewm reviewed (commented):
is `source` not an implementation detail of the task itself? I would expect that users would want to provide the path to the source within the git clone. The location of that source within the context should be irrelevant. 

i.e. https://github.com/konflux-ci/build-definitions/blob/main/task/tkn-bundle-oci-ta/0.1/tkn-bundle-oci-ta.yaml#L65-L66

In the buildah tasks, this is achieved with the use of a separate environment variable, `SOURCE_CODE_DIR`: https://github.com/konflux-ci/build-definitions/blob/main/task/buildah-oci-ta/0.4/buildah-oci-ta.yaml#L221-L222

avi-biton commented:
Thanks for the review @arewm, I will update the PR

avi-biton commented:
@arewm I added `SOURCE_CODE_DIR` parameter to the task with a default value of ""
In the pipeline definition I added this parameter with the value `source` 
PTAL and comment :)

chmeliik commented on task/tkn-bundle-oci-ta/0.1/tkn-bundle-oci-ta.yaml:
```diff
@@ -87,11 +93,11 @@ spec:
             path="${path#\!}"
           fi
-          if [[ -d "/var/workdir/source/${path}" ]]; then
+          if [[ -d "/var/workdir/source/${SOURCE_CODE_DIR}/${path}" ]]; then
```
I don't think this is correct for `oci-ta`. We download the source artifact like this

```
$(params.SOURCE_ARTIFACT)=/var/workdir/source
```

And the source artifact is just the repo, it doesn't have another `./source` subdirectory.

Most TA recipes do the replacement like this

https://github.com/konflux-ci/build-definitions/blob/4692a3811cf2cf319208729aa864b186f5f3743d/task/prefetch-dependencies-oci-ta/0.2/recipe.yaml#L51-L52

chmeliik replied:
Or like this, this may be better

https://github.com/konflux-ci/build-definitions/blob/4692a3811cf2cf319208729aa864b186f5f3743d/task/buildah-oci-ta/0.4/recipe.yaml#L13-L14

avi-biton replied:
Yes, I reverted the oci-ta to what it was and apply the changes only for `tkn-bundle`

avi-biton replied:
I now see that the CI is failing because its running the `generate-ta-tasks.sh` and builds the `tkn-bundle-oci-ta` task from the `tkn-bundle` task

avi-biton replied:
@chmeliik Can you PTAL.
I updated the recipe as in buildah task and updated the oci-ta task.

arewm commented on task/tkn-bundle-oci-ta/0.1/tkn-bundle-oci-ta.yaml:
```diff
@@ -28,6 +28,10 @@ spec:
+    - name: SOURCE_CODE_DIR
+      description: Source code directory in the working dir
+      type: string
+      default: /source
```
as a user, I would expect to input `source` over `/source`. Making this change would also require a change in the recipe to append the `/`.

avi-biton replied:
I could use the recipe to append the `/` but it will only append it to the `oci-ta` task and not the non ta task.
However I updated the parameter to be `source` and added the `/` in the environment variable for both versions of the task.

avi-biton commented on task/tkn-bundle-oci-ta/0.1/tkn-bundle-oci-ta.yaml:
```diff
@@ -60,7 +66,7 @@ spec:
         - $(params.SOURCE_ARTIFACT)=/var/workdir/source
```
Thanks @dheerajodha 
Its not something I can change manually, when adding `use-source` in the recipe, it will automatically generate it as /var/workdir/source when generating a ta task as defined in the `ta.go` file:
https://github.com/konflux-ci/build-definitions/blob/main/task-generator/trusted-artifacts/ta.go#L313
Do you think we can do it in another way?

avi-biton replied:
@arewm WDYT?

arewm replied:
If you cannot change the directory in the TA task, then that implies to me that the source code directory should be some subdirectory within that. Is the `source` directory not just where the git source is cloned?

Should a user provide the directory _within_ that cloned source directory ... therefore never having to worry about `source`? `source` definitely feels like an implementation detail now.

gbenhaim commented:
If there are no objections, I'll merge this PR tomorrow.

avi-biton commented:
/retest
